controllers/release/release_adapter.go

```diff
@@ -158,19 +161,39 @@ func (a *Adapter) EnsureReleasePipelineStatusIsTracked() (results.OperationResul
 	return results.ContinueProcessing()
 }
 
+// EnsureTargetContextIsSet is an operation that will ensure that the targetContext of the adapter is not nil. It will
+// set it to a context from the workspace in the ReleasePlan if one is defined or to its own context if not.
+func (a *Adapter) EnsureTargetContextIsSet() (results.OperationResult, error) {
+	releasePlan, err := a.getReleasePlan()
+	if err != nil {
+		return results.RequeueWithError(err)
+	}
+
+	if a.targetContext == nil {
```

Comment:
If you don't do anything when the `targetContext` has a value, the if above should be the check for the context being nil or not and then, you should do the rest. So:

1. Return if context is not nil
2. Get release plan
3. Set the proper value
```